src/inject/dynamic-theme/css-collection.ts

```diff
@@ -0,0 +1,62 @@
+import {StyleElement} from './style-manager';
+import {forEach} from '../../utils/array';
+
+function beautify(text: string) {
+    const CSS = (text
+        .replace(/\{/g,'{%--%') // {
+        .replace(/\}/g,'%--%}%--%') // }
+        .replace(/\;/g,';%--%') // ;
+        .replace(/%--%\s{0,}%--%/g,'%--%') // Remove %--% Without any characters between to the next %--%
+        .split('%--%'));
+    let deep = 0;
+    const formatted = [];
+    const shift = '   ';
+
+    for (let x = 0, len = CSS.length; x < len; x++) {
+        const line = CSS[x] + '\n';
+        if (line.match(/\{/)) { // {
+            formatted.push(shift.repeat(deep++) + line);
+        } else if (line.match(/\}/)) { // }
+            formatted.push(shift.repeat(--deep) + line);
+        } else { // CSS line
+            formatted.push(shift.repeat(deep) + line);
+        }
+    }
+    return formatted.join('');
+}
+
+export function exportCSSText() {
+    const css = [];
+
+    function addStaticCSS(selector: string, comment: string) {
+        const staticStyle = document.head.querySelector(selector);
+        if (staticStyle && staticStyle.textContent) {
+            css.push(`/* ${comment} */`);
+            css.push(beautify(staticStyle.textContent));
+            css.push('');
+        }
+    }
+
+    addStaticCSS('.darkreader--fallback', 'Fallback Style');
+    addStaticCSS('.darkreader--user-agent', 'User-Agent Style');
+    addStaticCSS('.darkreader--text', 'Text Style');
+    addStaticCSS('.darkreader--invert', 'Invert Style');
+    addStaticCSS('.darkreader--override', 'Override Style');
+    addStaticCSS('.darkreader--variables', 'Variables Style');
+
+    const modifiedCSS = [];
+    document.querySelectorAll('.darkreader--sync').forEach((element) => {
```

Comment:
You can use `element: HTMLStyleElement` type hint to avoid `as StyleElement`, because it is either `<style>` or `<link>`.